src/common/com/intellij/plugins/haxe/ide/index/HaxeComponentFileNameIndex.java

```diff
@@ -99,17 +101,26 @@ public static boolean getFileNames(@NotNull String qName,
     @NotNull
     public Map<String, Void> map(final FileContent inputData) {
       final PsiFile psiFile = inputData.getPsiFile();
+      final String moduleName = FileUtil.getNameWithoutExtension(inputData.getFileName());
       final List<HaxeClass> classes = HaxeResolveUtil.findComponentDeclarations(psiFile);
       if (classes.isEmpty()) {
         return Collections.emptyMap();
```

Comment:
I don't think you can leave this shortcut test and return here any more, since the code below is going to add the missing name to the result anyway. So, you should either return the missing class as the result, or, better yet, just let the algorithm run.